internal/ask/ask.go

```diff
@@ -0,0 +1,205 @@
+package ask
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+    "bufio"
+    "os"
+
+	aiapigo "github.com/inkeep/ai-api-go"
+	"github.com/inkeep/ai-api-go/models/components"
+	"github.com/inkeep/ai-api-go/models/sdkerrors"
+	"github.com/speakeasy-api/speakeasy/internal/charm/styles"
+
+	"github.com/speakeasy-api/speakeasy/internal/log"
+)
+
+type AskFlags struct {
+	Message string `json:"message"`
+	SessionID string `json:"sessionId,omitempty"`
+}
+
+
+var (
+    boldRegex   = regexp.MustCompile(`\*\*(.*?)\*\*`)
+    italicRegex = regexp.MustCompile(`\*(.*?)\*`)
+    linkRegex   = regexp.MustCompile(`\[\(?(.*?)\)?\]\((https?:\/\/[^\s]+)\)`)
+)
+
+
+func Ask(ctx context.Context, flags AskFlags) (string, error) {
+	logger := log.From(ctx)
+	s := aiapigo.New(aiapigo.WithSecurity(os.Getenv("INKEEP_API_KEY")))
+    var SessionID string
+	if flags.SessionID == "" {
+        res, err := s.ChatSession.Create(ctx, components.CreateChatSessionWithChatResultInput{
+            IntegrationID: os.Getenv("INKEEP_INTEGRATION_ID"),
+            ChatSession: components.ChatSessionInput{
+                Messages: []components.Message{{
+                    UserMessage: &components.UserMessage{
+                        Role:    "user",
+                        Content: flags.Message,
+                    },
+                }},
+            },
+        })
+		if err != nil {
+            SessionID = res.ChatResult.ChatSessionID
```

Comment:
I got a segfault here
Just running: `speakeasy ask`
And typing: `what are paths in openapi?` into the prompt

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x105457860]

goroutine 1 [running]:
github.com/speakeasy-api/speakeasy/internal/ask.Ask({0x1087b5280, 0x1400470f920}, {{0x140022b36a0?, 0x1?}, {0x0?, 0x0?}})
	/Users/robertcrumbaugh/code/speakeasy/internal/ask/ask.go:49 +0x270
github.com/speakeasy-api/speakeasy/internal/ask.RunInteractiveChatSession({0x1087b5280, 0x1400470f920}, {{0x0?, 0x1400470f920?}, {0x0?, 0x104a8b580?}})
	/Users/robertcrumbaugh/code/speakeasy/internal/ask/ask.go:178 +0x330
github.com/speakeasy-api/speakeasy/cmd.AskFunc({0x1087b5280?, 0x1400470f920?}, {{0x0?, 0x1054398c4?}, {0x0?, 0x0?}})
	/Users/robertcrumbaugh/code/speakeasy/cmd/ask.go:28 +0x2c
github.com/speakeasy-api/speakeasy/internal/model.ExecutableCommand[...].Init.func1.1()
	/Users/robertcrumbaugh/code/speakeasy/internal/model/command.go:95 +0x74
github.com/speakeasy-api/speakeasy/internal/model.ExecutableCommand[...].Init.func1.2(0x108213420?)
	/Users/robertcrumbaugh/code/speakeasy/internal/model/command.go:100 +0x2c
github.com/speakeasy-api/speakeasy-core/events.Telemetry({0x1087b5280, 0x1400470eff0}, {0x106554dba, 0x8}, 0x1400376b8d8)
	/Users/robertcrumbaugh/go/pkg/mod/github.com/speakeasy-api/speakeasy-core@v0.3.3/events/capture.go:79 +0x2dc
github.com/speakeasy-api/speakeasy/internal/model.ExecutableCommand[...].Init.func1({0x106536858, 0x4?, 0x10653684c?})
	/Users/robertcrumbaugh/code/speakeasy/internal/model/command.go:99 +0x1d8
github.com/spf13/cobra.(*Command).execute(0x1400544c608, {0x10a0149e8, 0x0, 0x0})
	/Users/robertcrumbaugh/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0x840
github.com/spf13/cobra.(*Command).ExecuteC(0x109d56640)
	/Users/robertcrumbaugh/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/robertcrumbaugh/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/speakeasy-api/speakeasy/cmd.Execute({0x10653bab8, 0x5}, {0x106571f73, 0xb})
	/Users/robertcrumbaugh/code/speakeasy/cmd/root.go:102 +0x138
main.main()
	/Users/robertcrumbaugh/code/speakeasy/main.go:11 +0x3c
exit status 2
```

Reply:
Did it fail because of the lack of environment variables? This is my best guess because I am not sure how environment variables are managed in the CLI so they're just placeholders at the moment (I'm used to Vercel but that doesn't seem applicable here).

Reply:
Oh wait I see what's up (but also yeah environment variables).